### ISSUE TITLE
Polyline Overlay with BitmapShader: strange visual effects

### ISSUE THREAD

**Issue by ludwigb** [mapsforge/mapsforge#578]
Laura reported that when using BitmapShaders with Polyline Overlays the line get some strange visual effects: 
> The code works, but the bitmap is not drawing in the
expected way, I want instead of draw line, I draw little shoes...the shoes
are drawing but in a weird way...and when you zoom enough the map and you
tap on the route line is like the shoes roll...extremely weird
behavior...
https://lists.openstreetmap.org/pipermail/graphhopper/2015-January/001599.html

The rolling effect of lines appears because the overlay bitmaps are drawn with a fixed alignment to the screen, while the underlying map moves. 
The solution will be to apply a shift to the Paint through the setBitmapShaderShift() method.

**Comment by ludwigb** [comment]
I just pushed a fix for this issue to dev with https://github.com/mapsforge/mapsforge/commit/906d854d0488fce297fe2b09e74660c33d51f0d2

The solution is to take the position of the screen and move the bitmap together with the underlying map. 
I have added a new ctor to Polyline that has an align parameter: when set to true, this alignment will applied, the default behaviour stays the same.

The same should probably also applied to the other Overlay forms (Polygon, Circle etc), waiting for feedback.
